### PR TITLE
Merge my bunch of fixes and additions

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
There may have been bugs introduced on the way, but didn't see those yet. Bukkit and bungeecord seem to work. Will need someone to test sponge and velocity.

### CODE REVIEW

File: src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java [hunk 1]
```diff
@@ -106,35 +94,31 @@ public String getVersion() {
...
+                YamlConfig mundoConfig = new YamlConfig(new File(getDataFolder().getParentFile(), "MundoSK"), "config.yml", false, srLogger);
                 mundoConfig.reload();
                 if (mundoConfig.getBoolean("enable_custom_skin_and_tablist")) {
-                    console.sendMessage("§e[§2SkinsRestorer§e] §4----------------------------------------------");
-                    console.sendMessage("§e[§2SkinsRestorer§e] §4             [CRITICAL WARNING]");
-                    console.sendMessage("§e[§2SkinsRestorer§e] §cWe have detected MundoSK on your server with §e'enable_custom_skin_and_tablist: §4§ntrue§e'§c.");
-                    console.sendMessage("§e[§2SkinsRestorer§e] §cThat setting is located in §e/plugins/MundoSK/config.yml");
-                    console.sendMessage("§e[§2SkinsRestorer§e] §cYou have to disable ('false') it to get SkinsRestorer to work!");
-                    console.sendMessage("§e[§2SkinsRestorer§e] §4----------------------------------------------");
+                    srLogger.info("§4----------------------------------------------");
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java:116]: Might be better to use warn here as this is critical info

File: src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java [hunk 2]
```diff
@@ -106,35 +94,31 @@ public String getVersion() {
...
+                    srLogger.info("§4             [CRITICAL WARNING]");
+                    srLogger.info("§cWe have detected MundoSK on your server with §e'enable_custom_skin_and_tablist: §4§ntrue§e'§c.");
+                    srLogger.info("§cThat setting is located in §e/plugins/MundoSK/config.yml");
+                    srLogger.info("§cYou have to disable ('false') it to get SkinsRestorer to work!");
+                    srLogger.info("§4----------------------------------------------");
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java:121]: Might be better to use WARN here as this is critical

File: src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java [hunk 3]
```diff
@@ -270,21 +250,11 @@ public void onEnable() {
...
-            checker.checkServices();
-            ServiceChecker.ServiceCheckResponse response = checker.getResponse();
-
-            if (response.getWorkingUUID() == 0 || response.getWorkingProfile() == 0) {
-                console.sendMessage("§c[§4Critical§c] ------------------[§2SkinsRestorer §cis §c§l§nOFFLINE§c] --------------------------------- ");
-                console.sendMessage("§c[§4Critical§c] §cPlugin currently can't fetch new skins due to blocked connection!");
-                console.sendMessage("§c[§4Critical§c] §cSee http://skinsrestorer.net/firewall for steps to resolve your issue!");
-                console.sendMessage("§c[§4Critical§c] ------------------------------------------------------------------------------------------- ");
-            }
+            SharedMethods.runServiceCheck(mojangAPI, srLogger);
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java:257]: Needs to be tested

File: src/main/java/net/skinsrestorer/bukkit/skinfactory/SkinFactory.java [hunk 4]
```diff
@@ -35,8 +36,16 @@
      * @param props - Property Object
      */
     default void applySkin(final Player p, Object props) {
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/bukkit/skinfactory/SkinFactory.java:38]: Need to be tested

File: src/main/java/net/skinsrestorer/bungee/SkinsRestorer.java [hunk 5]
```diff
@@ -233,22 +185,36 @@ private void checkUpdate(boolean showUpToDate) {
             public void updateAvailable(String newVersion, String downloadUrl, boolean hasDirectDownload) {
                 outdated = true;
 
-                updateChecker.getUpdateAvailableMessages(newVersion, downloadUrl, hasDirectDownload, getVersion(), false).forEach(msg ->
-                        console.sendMessage(TextComponent.fromLegacyText(msg)));
+                updateChecker.getUpdateAvailableMessages(newVersion, downloadUrl, hasDirectDownload, getVersion(), false).forEach(srLogger::info);
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/bungee/SkinsRestorer.java:188]: needs testing

Comment (AlexProgrammerDE) [reply src/main/java/net/skinsrestorer/bungee/SkinsRestorer.java:188]: This works.

File: src/main/java/net/skinsrestorer/shared/storage/Locale.java [hunk 6]
```diff
@@ -117,14 +118,14 @@ public static void load(String path) {
...
-                    if (!Arrays.stream(IGNORE_PREFIX).anyMatch(f.getName()::contains))
+                    if (Arrays.stream(IGNORE_PREFIX).noneMatch(f.getName()::contains))
                         parsed = C.c(locale.getString("PREFIX", null)) + parsed;
                 }
 
                 f.set(null, parsed);
             }
         } catch (Exception e) {
-            System.out.println("§e[§2SkinsRestorer§e] §cCan't read messages.yml! Try removing it and restart your server.");
+            logger.info("§cCan't read messages.yml! Try removing it and restart your server.");
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/shared/storage/Locale.java:128]: Should be warn

File: src/main/java/net/skinsrestorer/shared/storage/MySQL.java [hunk 7]
```diff
@@ -21,33 +21,28 @@
...
 public class MySQL {
+    private final SRLogger logger;
     private final String host;
     private final String port;
     private final String database;
     private final String username;
     private final String password;
     private final String options;
-    private final ExecutorService exe;
+    private final ExecutorService exe = Executors.newCachedThreadPool();
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/shared/storage/MySQL.java:43]: needs testing

File: src/main/java/net/skinsrestorer/shared/storage/MySQL.java [hunk 8]
```diff
@@ -68,30 +63,30 @@ public Connection openConnection() throws SQLException {
...
         try {
             if (con == null || !con.isValid(1)) {
-                System.out.println("[SkinsRestorer] MySQL connection lost! Creation a new one.");
+                logger.info("MySQL connection lost! Creation a new one.");
                 con = openConnection();
             }
         } catch (SQLException e) {
-            System.out.println("[SkinsRestorer] Could NOT connect to MySQL: " + e.getMessage());
+            logger.info("Could NOT connect to MySQL: " + e.getMessage());
         }
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/shared/storage/MySQL.java:78]: MYSQL error's should be warn

File: src/main/java/net/skinsrestorer/bukkit/skinfactory/SkinFactory.java (hunk 4 above)
Comment (AlexProgrammerDE) [reply src/main/java/net/skinsrestorer/bukkit/skinfactory/SkinFactory.java:38]: It still applies for me on bungeecord, so i think its fine.

**Pull request merged**